clients/client-py/pyproject.toml

```diff
@@ -0,0 +1,110 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "taskcluster"
+version = "91.0.1"
+description = "Python client for Taskcluster"
+readme = "README.md"
+license = {text = "MPL-2.0"}
+requires-python = ">=3.9"
+authors = [
+    {name = "Mozilla Taskcluster and Release Engineering", email = "release+python@mozilla.com"},
+]
+keywords = ["taskcluster", "api", "client"]
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3.13",
+    "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+
+dependencies = [
+    "requests>=2.4.3",
+    "mohawk>=0.3.4",
+    "python-dateutil>=2.8.2",
+    "slugid>=2",
+    "taskcluster-urls>=12.1.0",
+    "aiohttp>=3.7.4",
+    "async_timeout>=2.0.0",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest",
+    "pytest-cov",
+    "pytest-mock",
+    "pytest-asyncio",
+    "httmock",
+    "mock",
+    "flake8",
+    "psutil",
+    "hypothesis",
+    "coverage",
+    "aiofiles",
+    "httptest",
+]
+dev = [
+    "pytest",
+    "pytest-cov",
+    "pytest-mock",
+    "pytest-asyncio",
+    "httmock",
+    "mock",
+    "flake8",
+    "psutil",
+    "hypothesis",
+    "coverage",
+    "aiofiles",
+    "httptest",
+    "tox",
+    "ruff",
+    "grip",
+]
```

Comment:
You want these to be `dependency-groups`, not optional dependencies:
https://docs.astral.sh/uv/concepts/projects/dependencies/#development-dependencies

Optional dependencies are for consumers who want to install optional features and would allow people to do `pip install taskcluster[dev]` (which isn't what you want here).

Reply:
Also up to you, but I probably wouldn't bother having separate `dev` and `test` groups.